neuralforecast/tsdataset.py

```diff
@@ -370,13 +371,185 @@ def __init__(
+            # Add Available mask efficiently (without adding column to df)
+            if "available_mask" not in df.columns:
+                available_mask = np.ones((len(data), 1), dtype=np.float32)
+                data = np.append(data, available_mask, axis=1)
+                temporal_cols = temporal_cols.append(pd.Index(["available_mask"]))
+
+            data = self._as_torch_copy(data)
+
+            # Pad the temporal data to the left
+            temporal = torch.zeros(
+                size=(len(temporal_cols), self.files_ds.max_size), dtype=torch.float32
+            )
+            temporal[: len(temporal_cols), -len(data) :] = data.permute(1, 0)
+
+            # Add static data if available
+            static = None if self.static is None else self.static[idx, :]
+
+            item = dict(
+                temporal=temporal,
+                temporal_cols=temporal_cols,
+                static=static,
+                static_cols=self.static_cols,
+                y_idx=self.y_idx,
+            )
+
+            return item
+        raise ValueError(f"idx must be int, got {type(idx)}")
+
+    def __len__(self):
+        return self.n_groups
+
+    def _as_torch_copy(
+        self,
+        x: Union[np.ndarray, torch.Tensor],
+        dtype: torch.dtype = torch.float32,
+    ) -> torch.Tensor:
+        if isinstance(x, np.ndarray):
+            x = torch.from_numpy(x)
+        return x.to(dtype, copy=False).clone()
+
+    @staticmethod
+    def from_data_directory(
+        files,
+        static_df=None,
+        sort_df=False,
+        id_col="unique_id",
+        time_col="ds",
+        target_col="y",
+    ):
+        """We expect files_ds to have one parquet file per timeseries containing a pd df and static df to be a pd df"""
+
+        # Define indexes if not given
+        if static_df is not None:
+            if isinstance(static_df, pd.DataFrame) and static_df.index.name == id_col:
+                warnings.warn(
+                    "Passing the id as index is deprecated, please provide it as a column instead.",
+                    FutureWarning,
+                )
+            if sort_df:
+                static_df = ufp.sort(static_df, by=id_col)
+
+        # Static features
+        if static_df is not None:
+            static_cols = [col for col in static_df.columns if col != id_col]
+            static = ufp.to_numpy(static_df[static_cols])
+            static_cols = pd.Index(static_cols)
+        else:
+            static = None
+            static_cols = None
+
+        max_size = 0
+        min_size = float("inf")
+        temporal_cols = pd.Index([])
+        last_times = np.array([], dtype=np.datetime64)
+        indices = pd.Series([])
+
+        for file in files:
+            df = pd.read_parquet(file)
+            length = len(df)
+            if length > max_size:
+                max_size = length
+            if length < min_size:
+                min_size = length
```

Comment:
instead of if length > max_size can just do `max_size = max(max_size, length)`, and same for min_size